lib/grammers-client/src/client/client.rs

```diff
@@ -119,6 +119,14 @@ pub(crate) struct ClientInner {
     pub(crate) updates: Mutex<VecDeque<crate::types::Update>>,
     // Used to avoid locking the entire sender when enqueueing requests.
     pub(crate) request_tx: Mutex<Enqueuer>,
+    // Stores per-datacenter FileDownloader instances
+    pub(crate) downloader_map: RwLock<HashMap<i32, Arc<FileDownloader>>>,
+}
+
+pub(crate) struct FileDownloader {
```

Comment:
Not entirely sure if we should name these `FileDownloader`. They're more like generic exported senders after all (which are used for downloads, yes).

Reply:
what about ```DatacenterConnector``` ?

Reply:
Hmm not really... Perhaps we can leave it for now.